### PR TITLE
Proposal of multi-lingual programming mapper

### LINKED ISSUE #5594: [OSPP]  Multi-language mapper-framework support
**What would you like to be added/modified**:
 Implement Mapper-framework based on at least one language (c, java, python)

**Why is this needed**:
Mapper-framework All devices can be connected and controlled through the manufacturer's driver, but the information obtained from the device needs to be converted into a format understood by Kubeedge. Mapper is an interconnected application between Kubeedge and devices. Mapper-framework is a framework that makes writing mappers easier by providing a mapper runtime library and tools for scaffolding and code generation to generate new mappers. Currently, the community has Mapper-framework implemented based on the go language, but other languages ​​are also in demand in actual business.

### PR DESCRIPTION
**Pull request by ryWangkkk**
**What type of PR is this?**
/kind design

-->

**What this PR does / why we need it**:
[OSPP] Multi-language mapper-framework support
**Which issue(s) this PR fixes**:

Fixes #5594 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/proposals/multi-lingual programming mapper.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
+# Multi-lingual Programming Mapper
+
+## Table of Contents
```
Comment (wbc6080) [inline docs/proposals/multi-lingual programming mapper.md:3]: The proposal needs to add a header template. You can refer to the previous proposal like https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/mapper-design-v2.md?plain=1

File: docs/proposals/multi-lingual programming mapper.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+5) **Http** is responsible for implementing the interface for pushing data to third-party applications.
+6) **Model, Service** is responsible for defining complex struct variables and interfaces separately.
+
+### Implementation
+
+```
+mapper
+├── src
+│ └── main
+│  └── java
```
Comment (wbc6080) [inline docs/proposals/multi-lingual programming mapper.md:181]: I have a question. According to your design, will you eventually provide a java version of mapper-framework that can automatically generate a java version of mapper project?

File: docs/proposals/multi-lingual programming mapper.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+maintains the list of devices on the Mapper, and updates device expectations.
+3) **DevicePanel** module creates a DevPanel to manage the device lifecycle.
+4) **Driver module** reads and writes device data and converts it through the protocol.
+5) **DevicePanel** module gets and sets data.
+6) Meanwhile, **Data** module processes data in streaming or non-streaming modes,
+7) and stores it in a database 
+8) or pushes it to third-party apps.
+9) The Mapper reports device status and data back to EdgeCore using **GrpcClient** module.
+
+### Component
```
Comment (wbc6080) [inline docs/proposals/multi-lingual programming mapper.md:47]: This part mainly explains some modifications made at the mapper level, but we should also make some modifications in the dmi api of devicetwin. Now we have implemented a grpc dmi interface and let it automatically generate the go version of the interface file in https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/dmi/v1beta1/api.proto. In this design, it should also be necessary to let grpc's dmi interface generate a Java version of the interface file. You can add a description of this part

File: docs/proposals/multi-lingual programming mapper.md (hunk 1 above)
Comment (ryWangkkk) [reply docs/proposals/multi-lingual programming mapper.md:3]: OK, I‘ll add a header later.

File: docs/proposals/multi-lingual programming mapper.md (hunk 2 above)
Comment (ryWangkkk) [reply docs/proposals/multi-lingual programming mapper.md:181]: Of course! In fact,there should be a Makefile in this design. After developer finish their own protocol, they could use the Makefile to generate a java version of mapper project. But I accidentally made a clerical mistake and wrote the Makefile incorrectly, resulting in two Dockerfiles. The actual file schema should look like the following diagram. I will correct it later.
微信图片_20240801120359

File: docs/proposals/multi-lingual programming mapper.md (hunk 3 above)
Comment (ryWangkkk) [reply docs/proposals/multi-lingual programming mapper.md:47]: In fact, [https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/dmi/v1beta1/api.pb.go](https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/dmi/v1beta1/api.pb.go) and [https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/dmi/v1beta1/api_grpc.pb.go](https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/dmi/v1beta1/api_grpc.pb.go)  are both needed in the mapper project. However, there is too much code(3964 lines in api.pb.go) in these files, and not all is necessary in the mapper project. So we  choose to rewrite those indispensable code by java in this project. If needed, I could add this to "Component" and explain it in detail.
微信图片_20240801123102

**Review by wbc6080 — COMMENTED** [review]
/lgtm
cc @fisherxu @WillardHu

**Review by fisherxu — COMMENTED** [review]
/approve
/assign @WillardHu

**Review by Shelley-BaoYue — COMMENTED** [review]
/lgtm

**Pull request merged**